project/src/components/map/map.tsx

```diff
@@ -1,11 +1,58 @@
 import cn from 'classnames';
+import React, { useRef, useEffect } from 'react';
+import { Icon, Marker } from 'leaflet';
+import 'leaflet/dist/leaflet.css';
+import { Offer } from '../../types/Offer';
+import useMap from '../../hooks/useMap';
+import { URL_MARKER_CURRENT, URL_MARKER_DEFAULT } from '../../const';
 
 type MapProps = {
   className: string;
+  offers: Offer[];
+  selectedPoint?: Offer | undefined;
 }
 
-const Map = ({ className }: MapProps): JSX.Element => (
-  <section className={cn('map', className)} />
-);
+const defaultCustomIcon = new Icon({
+  iconUrl: URL_MARKER_DEFAULT,
+  iconSize: [40, 40],
+  iconAnchor: [20, 40]
+});
+
+const currentCustomIcon = new Icon({
+  iconUrl: URL_MARKER_CURRENT,
+  iconSize: [40, 40],
+  iconAnchor: [20, 40]
+});
+
+const Map = ({ className, selectedPoint, offers }: MapProps): JSX.Element => {
+  const mapRef = useRef(null);
+  const map = useMap(mapRef, offers[0]);
+
+  useEffect(() => {
+    if (map) {
+      offers.forEach((offer) => {
+        const marker = new Marker({
+          lat: offer.city.location.latitude,
+          lng: offer.city.location.longitude
+        });
+
+        marker
+          .setIcon(
+            selectedPoint !== undefined && offer.id === selectedPoint.id
```

Comment:
Для сравнения нам достаточно айди, зачем в стейт всю точку хранить?
```suggestion
            selectedPointId && offer.id === selectedPointId
```